datajs/src/file-base.js

```diff
@@ -140,13 +142,35 @@ export class FileInterface extends File {
     return this.descriptor.name
   }
 
-  async hash() {
-    const text = await this.descriptor.text()
-    return crypto.createHash('md5').update(text).digest('hex')
-  }
-
-  async hashSha256() {
-    const text = await this.descriptor.text()
-    return crypto.createHash('sha256').update(text).digest('hex')
+  async generateHash(hashType, cbProgress) {
```

Comment:
Can you add JS doc comments, describing each param? Like we have for other functions.